usagov/usagov-2021
# USAGOV-1770 filter out search.gov

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1770

## Description

Modifies the usa.json so that "/search" results are filtered out of the top 7 and 30 day results. Adds a test script for easier validation. 

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

1.  Get environment variables and newkey.json files from Isabel if you do not already have them 
2. You'll also need the docker daemon running
3. In localreport.sh: 
    1. replace exports from the values in environment file
4. Add "newkey.jon" to the analytics-reporter directory (note this is one directory down from local report.sh) 
5. In Dockerfile-reporter 
    1. on line 4 replace COPY .docker/src-reporter . with COPY src-reporter .
    2. on line 13 replace  RUN chmod +x /createreport.sh \ with RUN chmod +x /localreport.sh \
    3. on line 20 replace CMD ["./createreport.sh"] with CMD ["./localreport.sh"]
6. Navigate to /usagov-2021/.docker  and run: 
    1. docker context use default 
    2. docker build -t analytics_reporter_test -f Dockerfile-reporter . 
        1. **Note the ending period 
    3. docker run analytics_reporter_test
7. In the results check if you get a pagePath="/search" (you shouldn't) 
8. To validate further you can go to the file /analytics-reporter/reports/usa.json
    1. Scroll down to the "top-pages-7-days" (2nd to last entry) and remove the "dimensionFilter" (lines 589 to 605) 
    2. If you remove this and rerun it you should see a pagePath="/search"
    3. You can also test this changing it in the "top-pages-30-days" or changing ANALYTICS_REPORT_IDS to ANALYTICS_REPORT_ENGLISH or ANALYTICS_REPORT_SPANISH

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
